### ISSUE TITLE
CW attack cannot start on the FaceNet example

### ISSUE THREAD

**Issue by ywu36** [tensorflow/cleverhans#1052]
**Describe the bug**
When I'm trying to use the CW attack on the FaceNet example (https://github.com/tensorflow/cleverhans/blob/master/examples/facenet_adversarial_faces/facenet_fgsm.py), I used the following codes to replace the FGSM attack:

    cw =  CarliniWagnerL2(model, sess=sess)
    fgsm_params = {
                   'clip_min': 0.,
                   'clip_max': 1.}
    model.face_input.set_shape(np.shape(faces2))
    adv_x = cw.generate(model.face_input, **fgsm_params)
    feed_dict = {model.face_input: adv,
                   model.victim_embedding_input: victims_embeddings,
                   phase_train_placeholder: False}
    adv = sess.run(adv_x, feed_dict=feed_dict)

It reports an error: "InvalidArgumentError (see above for traceback): You must feed a value for placeholder tensor 'batch_size' with dtype int32".

InvalidArgumentError (see above for traceback): You must feed a value for placeholder tensor 'batch_size' with dtype int32
 [[node batch_size (defined at cleverhans-master/examples/facenet_adversarial_faces/facenetDir/src/facenet.py:373)  = Placeholder[dtype=DT_INT32, shape=<unknown>, _device="/job:localhost/replica:0/task:0/device:CPU:0"]()]]
 [[node PyFunc (defined at ~/anaconda36/lib/python3.6/site-packages/cleverhans/attacks/carlini_wagner_l2.py:82)  = PyFunc[Tin=[DT_FLOAT, DT_FLOAT], Tout=[DT_FLOAT], token="pyfunc_0", _device="/job:localhost/replica:0/task:0/device:CPU:0"](_arg_input_0_1, StopGradient/_941)]]

I thought it might because the FaceNet model is an external model, after importing, the shape of its input tensor became UnKnown. However, after I manually assign its a shape: "model.face_input.set_shape(np.shape(faces2))", the error still exists. Not sure why. 

**System configuration**
 - OS: Linux Redhat
 - Python version: Python 3.6
 - TensorFlow version: 1.12

**Comment by B-Yassine** [comment]
I resolved the same issue by setting the learning phase to false and i really don't know why it works. 
Using Keras backend:

```
from keras import backend as K

K.set_learning_phase(False)
```

**Comment by npapernot** [comment]
That's probably because it disables something like dropout or batch norm that requires a value for a placeholder that was missing from the session run. Thanks for the response

**Comment by luannd** [comment]
@npapernot @carlini I still got the same error as @ywu36 .The above solution doesn't work in TF. I can run the cw attack on MNIST but don't know why it fails on facenet.

**Comment by jpgard** [comment]
Thanks for the answer @B-Yassine ! And thank you to all the devs who contributed to cleverhans.

I will add that I was doing experiments with a custom pretrained Keras model in cleverhans, and also encountered the `You must feed a value for placeholder tensor` error. The error was raised specifically at the `model.compile()` call and was due to something in the call to the custom `adv_acc` metric which I was using exactly as in `mnist_tutorial_keras.py`. 

While eventually I discovered this post and was able to use `K.set_learning_phase(False)` to solve the problem, it would be very useful if cleverhans would provide some kind of useful error message to help point to this issue. I spent hours looking for a solution and this never would have occurred to me (indeed, it is not even clear why it works). I'll also note that this issue only seems to affect custom models loaded from elsewhere, as it does not affect the `cnn_model()` from the tutorial, which I was able to run without issue.
